crates/revmc-worker/src/handler.rs

```diff
@@ -1,48 +1,46 @@
-use std::{ panic::{ catch_unwind, AssertUnwindSafe }, sync::{ Arc, Mutex } };
+use std::{
+    panic::{catch_unwind, AssertUnwindSafe},
+    sync::Arc,
+};
 
-use revm::{ handler::register::EvmHandler, Database };
+use revm::{handler::register::EvmHandler, Database};
 
 use crate::EXTCompileWorker;
 
 // Register handler for external context to support background compile worker in node runtime
+//
+// Placeholders for handling unexpected behaviors are left to be placed by developers
 pub fn register_handler<DB: Database + 'static>(
-    handler: &mut EvmHandler<'_, Arc<Mutex<EXTCompileWorker>>, DB>
+    handler: &mut EvmHandler<'_, Arc<EXTCompileWorker>, DB>,
 ) {
     let prev = handler.execution.execute_frame.clone();
     handler.execution.execute_frame = Arc::new(move |frame, memory, tables, context| {
         let interpreter = frame.interpreter_mut();
         let code_hash = interpreter.contract.hash.unwrap_or_default();
         let spec_id = context.evm.inner.spec_id();
-        let function_result = context.external.lock().unwrap().get_function(code_hash);
-        match function_result {
+
+        match context.external.get_function(code_hash) {
             Ok(None) => {
                 let bytecode = context.evm.db.code_by_hash(code_hash).unwrap_or_default();
-                context.external
-                    .lock()
-                    .unwrap()
-                    .work(spec_id, code_hash, bytecode.original_bytes());
 
+                if let Err(_err) =
+                    context.external.work(spec_id, code_hash, bytecode.original_bytes())
+                {
+                };
```

Comment:
we don't have to handling error from exteranl.work?

Reply:
it should be done in revm-wise, since register handler is being called in revm
previously just logged it with tracer, but since we don't use tracer, i think it's okay to leave it as a placeholder

Reply:
okay i'll add error log to trace